code/game/gamemodes/dynamic/antag_rulesets.dm

```diff
@@ -0,0 +1,191 @@
+/**
+ * These are gamemode rulesets for the dynamic gamemode type. They determine what antagonists spawn during a round.
+ */
+/datum/ruleset
+	/// What this ruleset is called
+	var/name = "BASE RULESET"
+	/// The cost to roll this ruleset
+	var/ruleset_cost = 1
+	/// The weight to roll this ruleset
+	var/ruleset_weight = 1
+	/// The cost to roll an antagonist of this ruleset
+	var/antag_cost = 1
+	/// The weight to roll an antagonist of this ruleset
+	var/antag_weight = 1
+	/// Antagonist datum to apply to users
+	var/datum/antagonist/antagonist_type
+	/// A ruleset to be added when this ruleset is selected by the gamemode
+	var/datum/ruleset/implied/implied_ruleset_type
+
+	/// These roles 100% cannot be this antagonist
+	var/list/banned_jobs = list("Cyborg")
+	/// These roles can't be antagonists because mindshielding (this can be disabled via config)
+	var/list/protected_jobs = list(
+		"Security Officer",
+		"Warden",
+		"Detective",
+		"Head of Security",
+		"Captain",
+		"Blueshield",
+		"Nanotrasen Representative",
+		"Magistrate",
+		"Internal Affairs Agent",
+		"Nanotrasen Career Trainer",
+		"Nanotrasen Navy Officer",
+		"Special Operations Officer",
+		"Trans-Solar Federation General"
+	)
+	/// Applies the mind roll to assigned_role, preventing them from rolling a normal job. Good for wizards and nuclear operatives.
+	var/assign_job_role = FALSE
+	/// A blacklist of species names that cannot play this antagonist
+	var/list/banned_species = list()
+	/// If true, the species blacklist is now a species whitelist
+	var/banned_species_only = FALSE
+
+	// var/list/banned_mutual_rulesets = list() // UNIMPLEMENTED: could be used to prevent nukies rolling while theres cultists, or wizards, etc
+
+	/* This stuff changes, all stuff above is static */
```

Comment:
Don't we have final vars or something in spacemandmm? Might not be a bad idea to use that

Reply:
Im pretty sure it means you cant override it on subtypes if you do it that way